library/alloc/src/string.rs

```diff
@@ -2081,7 +2081,31 @@ impl FromUtf8Error {
     #[cfg(not(no_global_oom_handling))]
     #[unstable(feature = "string_from_utf8_lossy_owned", issue = "129436")]
     pub fn into_utf8_lossy(self) -> String {
-        String::from_utf8_lossy_owned(self.bytes)
+        const REPLACEMENT: &str = "\u{FFFD}";
+
+        let mut res = {
+            let mut v = Vec::with_capacity(self.bytes.len());
+
+            // `Utf8Error::valid_up_to` returns the maximum index of validated
+            // UTF-8 bytes. Copy the valid bytes into the output buffer.
+            v.extend(self.bytes[..self.error.valid_up_to()].iter().copied());
```

Comment:
can you use `extend_from_slice` here? that's both simpler and possible more efficient (though the iterator probably produces good code as well? but either way, extend_from_slice is shorter)

Reply:
Sure, I normally use `extend_from_slice` and second-guessed myself since I couldn't remember if there was a difference.
I checked the docs and went with this because of the last sentence here but that's maybe a bit hopeful still.
> Note that this function is same as [`extend`](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.extend) except that it is specialized to work with slices instead. If and when Rust gets specialization this function will likely be deprecated (but still available).

https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.extend_from_slice